nixos/modules/services/misc/adaptivemm.nix

```
@@ -0,0 +1,86 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adaptivemm;
+in {
+  options = {
+    services.adaptivemm = {
+      enable = mkEnableOption (lib.mdDoc "Proactively tune kernel free memory configuration");
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.adaptivemm;
+        defaultText = literalExpression "pkgs.adaptivemm";
+        description = lib.mdDoc ''
+          Which adaptivemm package to use.
+        '';
+      };
+
+      verbosity = mkOption {
+        type = types.int;
+        default = 0;
+        defaultText = literalExpression "0";
+        description = lib.mdDoc ''
+          Verbosity, from 0 to 5.
+        '';
+      };
+
+      maxWatermarkGap = mkOption {
+        type = types.int;
+        default = 5;
+        defaultText = literalExpression "5";
+        description = lib.mdDoc ''
+          Maximum allowed gap between high and low watermarks in GB.
+        '';
+      };
+
+      aggressiveness = mkOption {
+        type = types.int;
+        default = 2;
+        defaultText = literalExpression "2";
+        description = lib.mdDoc ''
+          Aggressiveness level (1=high, 2=normal, 3=low).
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment = {
+      systemPackages = [ cfg.package ];
+      etc."sysconfig/adaptivemmd".source = "${cfg.package}/etc/sysconfig/adaptivemmd";
+    };
+
+    systemd.services.adaptivemmd = {
+      serviceConfig = {
+        Type = "forking";
+        EnvironmentFile = "/etc/sysconfig/adaptivemmd";
```

unless you have a strong suspicion that it would be beneficial to the average user to run this manually, I think I would just hide this file and keep the package away from the environment.

```suggestion
    systemd.services.adaptivemmd = {
      serviceConfig = {
        Type = "forking";
        EnvironmentFile = "${cfg.package}/etc/sysconfig/adaptivemmd";
```